vllm/outputs.py

```diff
@@ -135,26 +143,50 @@ def from_seq_group(cls, seq_group: SequenceGroup) -> "RequestOutput":
         # NOTE: We need omit logprobs here explicitly because the sequence
         # always has the logprobs of the sampled tokens even if the
         # logprobs are not requested.
-        include_logprobs = seq_group.sampling_params.logprobs is not None
-        text_buffer_length = seq_group.sampling_params.output_text_buffer_length
-        outputs = [
-            CompletionOutput(
-                seqs.index(seq),
-                seq.get_output_text_to_return(text_buffer_length),
-                seq.data._output_token_ids,
-                seq.get_cumulative_logprob() if include_logprobs else None,
-                seq.output_logprobs if include_logprobs else None,
-                SequenceStatus.get_finished_reason(seq.status),
-                seq.stop_reason) for seq in top_n_seqs
-        ]
+        include_logprobs = sampling_params.logprobs is not None
+        text_buffer_length = sampling_params.output_text_buffer_length
+        deltas = sampling_params.output_kind == RequestOutputKind.DELTA
+
+        outputs = []
+        include_prompt = True
+        for seq in top_n_seqs:
+            output_text = seq.get_output_text_to_return(text_buffer_length)
+            output_logprobs = seq.output_logprobs if include_logprobs else None
+
+            output_token_ids: GenericSequence[int]
+            if deltas:
+                output_token_ids = seq.data.last_appended_tokens
+                seq.data.last_appended_tokens = []
```

Comment:
I would use .clear() here since it saves on deallocation=>allocation.

Reply:
But wouldn't this clear the list that we're setting in the `CompletionOutput` object that's getting returned?

Reply:
Resolving since this is no longer applicable after latest refactor.